bin/README.md

```diff
@@ -0,0 +1,466 @@
+lite: core 4bb376f0
+lite: final state 4bb376f0
+eval (run):
+4
+```
+The result (i.e. `4`) is printed to `stdout`. If the command had failed to
+compile, the stack trace would have been printed to `stdout` instead. All other
+messages are printed to `stderr`.
+
+#### Options
+
+`eval` was extended with several options that make it useful for processing Hoon
+nouns as input to or output from `conn.c`:
+- `-j`, `--jam`: output result as a jammed noun
+- `-c`, `--cue`: read input as a jammed noun
+- `-n`, `--newt`: write output / read input as a newt-encoded jammed noun, when
+  paired with `-j` or `-c` respectively
+- `-k`: treat the input as the jammed noun input of a `%fyrd` request to
+  `conn.c`; if the result is a `goof`, pretty-print it to `stderr` instead of
+  returning it
+
+### `-eval` and `-khan-eval`
+
+Two threads that evaluate arbitrary Hoon were added to the suite of threads
+included with Arvo:
+[`ted/eval.hoon`](https://github.com/urbit/urbit/blob/develop/pkg/arvo/ted/eval.hoon)
+and
+[`ted/khan-eval.hoon`](https://github.com/urbit/urbit/blob/develop/pkg/arvo/ted/khan-eval.hoon).
+
+Both threads take the same input: Hoon code as a `cord` and an optional
+`(list path)`. The optional `(list path)` is a list of Clay file dependencies
+which need to be included for the Hoon to be evaluated (i.e. if the Hoon code
+includes libraries or types defined outside of the kernel). Each `path` can be a
+`beam` (i.e. `[beak spur]`) or just a `spur`, in which case the default `beak`
+(i.e. `[our %base now]`) will be prepended.
+
+`ted/eval.hoon` expects the input to be a Hoon expression. It's very similar to
+`./urbit eval`, except that it has access to ship state: `now`, `our`, vane &
+agent state, etc.
+
+`ted/khan-eval.hoon` expects the input to be a thread. It attempts to compile
+the thread using the dependencies (if any) and then sends it to Khan as a `%lard`
+thread request.
+
+Both threads return regular thread output, i.e. a `vase`.
+
+Examples:
+- `-eval '(add 2 2)'`
+- `-eval '(my-add 2 2)' [/lib/my-add/hoon ~]`
+    - Where `my-add` is defined in `lib/my-add.hoon` in `%base`
+- `-eval '(my-add 2 2)' [/(scot %p our)/my-desk/(scot %da now)/lib/my-add/hoon ~]`
+    - Where `my-add` is defined in `lib/my-add.hoon` in `%my-desk`
+- `-khan-eval '=/  m  (strand ,vase)  ;<  ~  bind:m  (poke [~zod %hood] %helm-hi !>(\'\'))  (pure:m !>(\'success\'))'`
+
+### click
+
+[click](https://github.com/urbit/vere/blob/develop/bin/click) is a `bash` thin
+client which auto-formats `-eval` and `-khan-eval` thread calls via `%fyrd`
+requests to `conn.c` and coordinates chaining together the appropriate commands
+to execute those requests on a running ship.
+
+Using click, a call like:
+```
+echo $'[0 %fyrd %base %khan-eval %noun %ted-eval \'=/  m  (strand ,vase)  ;<  ~  bind:m  (poke [~zod %hood] %helm-hi !>(\\\'\\\'))  (pure:m !>(\\\'success\\\'))\']' | ./test-urbit eval -jn | nc -U -W 1 ~/Urbit/test-piers/zod/.urb/conn.sock | ./test-urbit eval -cn
```

Comment:
I think this would be more readable if split over multiple lines, but maybe you considered that and are optimizing instead for copy-pastability.

Reply:
I agree that the examples would be a lot more readable if split. Why not split on `|`? You can terminate with it to avoid backslashes.

Also, @ashelkovnykov looks like you have a couple `./test-urbit` instances here instead of `./urbit`

Reply:
@mcevoypeter Yeah, I was going for "easiest to copy-paste", but I think you two might be right about splitting the commands along the pipes

@barter-simsum Good catch, will change those